### ISSUE TITLE
CKA

### ISSUE THREAD

**Issue by charonooooooo** [dslisleedh/ESC#8]
Hello, thanks for your excellent work! I would like to ask how you generated the CKA

**Comment by dslisleedh** [comment]
Thank you for your interest, @charonooooooo!

We visualized CKA similarities using [this code](https://github.com/jayroxis/CKA-similarity/blob/main/CKA.py). 
To leverage the batched features, we cropped each Urban100x4 image to its smallest spatial size.

Please let me know if you have any further questions!

**Comment by charonooooooo** [comment]
> Sorry for the confusion [@charonooooooo](https://github.com/charonooooooo) .
> 
> The batch denotes the whole 100 samples of urban100x4, and spatial size denotes the smallest number of pixels on each h and w across 100 samples to concatenate 100 samples input-wise. I think the code below will be clear.
> 
> def load_imgs2(img_paths):
>     imgs = []
>     min_w, min_h = 10000, 10000
>     for img_path in img_paths:
>         img = Image.open(img_path).convert('RGB')
>         w, h = img.size
>         min_w = min(min_w, w)
>         min_h = min(min_h, h)
>         img = torch.tensor(np.array(img).transpose(2, 0, 1)).unsqueeze(0).float() / 255.0
>         imgs.append(img)
> 
>     imgs = [
>         img[..., :min_h, :min_w]
>         for img in imgs
>     ]
>     imgs = torch.cat(imgs, dim=0)
>     print(imgs.shape)  # torch.Size([100, 3, 143, 141])
>     return imgs

Thank you! The matter has been resolved！

**Comment by charonooooooo** [comment]
<img width="2037" height="1736" alt="Image" src="https://github.com/user-attachments/assets/4b2c9010-4764-4b1b-b41a-a33d19e491b6" />Hello author, this is the CKA similarities I drew, which is different from the one in the paper. The input of CKA is two-dimensional, but the dimension of features is three-dimensional. How do you solve this problem?

**Comment by charonooooooo** [comment]
<img width="2037" height="1750" alt="Image" src="https://github.com/user-attachments/assets/fe4b2234-2af8-43ab-89e5-40bc7651d1fa" />The result drawn according to the above code is still different from the paper, and it looks the same as my previous drawing.

**Comment by dslisleedh** [comment]
<img width="738" height="781" alt="Image" src="https://github.com/user-attachments/assets/21e9d541-de6e-42ba-81ff-9a52b4c860ba" />

Hmm, I run the code and get the same result as yours, @charonooooooo .
Actually, this result exhibits higher similarity than the result from the paper.
Perhaps I accidentally used results calculated from different datasets or different models. 

I'll update the figure in the arXiv version.
Thank you for your feedback, @charonooooooo !
